### PR TITLE
Redesigning the error and loading states in the reference demo

### PR DESCRIPTION
**Pull request by erunion**
This introduces a few minor redesigns to our https://preview.readme.io/reference/ page and introduces:

* A loading state.
* Handling error states a little differently and removing the "Like what you see?" box if errors are present.

#### Loading state
Screen Shot 2019-07-02 at 2 32 31 PM

#### Better error handling
Old:
Screen Shot 2019-07-02 at 2 30 57 PM

Refreshed:
Screen Shot 2019-07-02 at 2 30 18 PM

### CODE REVIEW

File: example/reference/src/Preview.jsx [hunk 1]
```diff
@@ -14,33 +14,57 @@ class Preview extends React.Component {
...
-            }
+              ) : (
+                <div className="hub-reference-section">
+                  <div id="hub-reference-loading-error">
+                    <div className="hub-reference-loading-error-box">
+                      <h3 className="hub-reference-loading-error-header">
+                        Couldn&apos;t be loaded
+                      </h3>
+                      <p>
+                        TKTK Lorem ipsum dolor sit amet, consectetur adipiscing elit TKTK
```
Comment (erunion) [inline example/reference/src/Preview.jsx]: @mjcuva Need copy here

Comment (mjcuva) [reply example/reference/src/Preview.jsx]: How about: "There was an error fetching your API Specification"

**Comment by erunion** [comment]
@mjcuva This is ready for review again!

**Comment by mjcuva** [comment]
If there is a validation error , can we move it inside the "Couldn't be loaded" section and change the copy to be "Invalid API Specification" with the error beneath it and a link to the validation report?

Similar to this (but we can leave the same design we currently have): 
image

**Comment by erunion** [comment]
@mjcuva @rafegoldberg This is ready for review again. I've revised the error states on the preview site to no longer show our debugging "status" text (while retaining it on the demo site since that's more for Explorer development).

### Preview site
#### Unable to parse a spec
Screen Shot 2019-07-25 at 9 49 33 AM

#### Unable to fetch a spec
Screen Shot 2019-07-25 at 9 49 48 AM

### Demo site
#### Unable to parse a spec
Screen Shot 2019-07-25 at 9 49 37 AM

#### Unable to fetch a spec
Screen Shot 2019-07-25 at 9 49 54 AM

**Pull request merged**